thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/Pipeline.java

```diff
@@ -0,0 +1,39 @@
+package com.linkedin.thirdeye.rootcause;
+
+import java.util.Set;
+
+
+/**
+ * Interface for a named stateless pipeline as injected into {@code RCAFramework}. Holds the business logic for
+ * associating search context entities with other relevant entities. Also performs relative ranking
+ * of associated entities in terms of importance to the user.
+ *
+ * @see RCAFramework
+ */
+public abstract class Pipeline {
+  final String name;
+  final Set<String> inputs;
+
+  public Pipeline(String name, Set<String> inputs) {
```

Comment:
please name the variables accordingly. Better yet have a concrete class to represent PipelineName